.github/workflows/cd.yml

```diff
@@ -0,0 +1,67 @@
+name: Continuous Deployment
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group:  ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  cd:
+    name: CD
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        lfs: false
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.8'
+
+    - name: Cache dependency
+      id: cache-dependencies
+      uses: actions/cache@v4
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .[dev]
+
+    - name: Install and run pre-commit hooks
```

Comment:
we had this separate before using the the pre-commit action which doesn't require installing any dependencies. why move this back in?

Reply:
I am not sure if I fully understood your question here. So in the current test, we are requesting a container twice. This is totally available. There is a slight preference for this IMO (even though it is not what the official action example does) since it repeats exactly what happens when a user installs everything. Imagine a (rare) case where for some reason the dependencies of sbi cause a problem with pre-commit. We would only notice this locally if we did it in two different containers.
I am totally open for changing it back though.